covariance/covarianceXsec.cpp

```diff
@@ -94,59 +112,321 @@ covarianceXsec::covarianceXsec(const char *name, const char *file, double thresh
   xsec_stepscale_vec.resize(nPars);
 
   for (int i = 0; i < nPars; i++) {
-    // Fill the nominal
-    xsec_param_nom_a[i] = (*xsec_param_nom)(i);
+    // Fill the prior central value
+	xsec_param_nom_a[i] = _fPreFitValue[i];
+
     // Fill the lower bound
-    xsec_param_lb_a[i]  = (*xsec_param_lb)(i);
+    xsec_param_lb_a[i]  = _fLowBound[i];
+
     // Fill the upper bound
-    xsec_param_ub_a[i]  = (*xsec_param_ub)(i);
-    // Fill the prior
-    xsec_param_prior_a[i]=(*xsec_param_prior)(i);
+    xsec_param_ub_a[i]  = _fUpBound[i];
+    // Fill the prior uncertainty
+	xsec_param_prior_a[i]= _fError[i];
     // Fill the names
-    xsec_param_names.push_back(std::string(((TObjString*)objarr_name->At(i))->GetString()));
+    xsec_param_names.push_back(_fNames[i]);
+
+    if(xsec_param_names[i].length() > PrintLength) PrintLength = xsec_param_names.back().length();
     // DB Fill the stepscales vector
-    xsec_stepscale_vec[i] = (*xsec_stepscale)(i);    
+    xsec_stepscale_vec[i] = _fIndivStepScale[i];
 
+    if(_fFlatPrior[i]){setEvalLikelihood(i, true);}
+    
     for (int j = 0; j < ncols; j++) {
       xsec_param_id_a[i][j] = (*xsec_param_id)(i,j);
```

Comment:
`xsec_param_id` is not initialized anymore, as L73 is commented. I am not sure how this should be resolved, but it is unusable in its current state.